### PR TITLE
Add test cases for multi-line quoted string

### LINKED ISSUE #83: Parsing fails when trying to parse a multi-line quoted string
This is valid in Lua 5.1:
```lua
local foo = "bar\
baz"
```

But full-moon panics when it comes across something like this. It shouldn't do that and instead should parse it properly.

### PR DESCRIPTION
**Pull request by zummenix**
~~I want to tackle the #83. So far I've added test cases for that, they are failing right now.~~

~~My next step is to add code to remove hanging `\` at the end of a line in `advance_quote` function of `tokenizer.rs` and save a string in `Token::StringLiteral` as new `Cow::Owned`. What do you think?~~

Closes #83

### CODE REVIEW

File: full-moon/src/tokenizer.rs [hunk 1]
```diff
@@ -1263,6 +1263,24 @@ mod tests {
             }))
         );
 
+        test_advancer!(
+            advance_quote(
+                r#""hello\
+world""#
+            ),
```
Comment (Kampfkarren) [inline full-moon/src/tokenizer.rs:1270]: Can you indent the `world` to be on the same line?

**Comment by Kampfkarren** [comment]
> My next step is to add code to remove hanging \ at the end of a line in advance_quote function of tokenizer.rs and save a string in Token::StringLiteral as new Cow::Owned. 

I'm not sure what you mean by this, can you elaborate?

**Comment by zummenix** [comment]
@Kampfkarren 

I've made tests to pass. As you can see the problem is that it leaves `\` in the string.

**Comment by Kampfkarren** [comment]
I think it should keep `\` in the string, no? full-moon is intended to be lossless.

It's possible even that the issue report was wrong if you can get these tests to pass without fiddling with the parser at all.

**Comment by zummenix** [comment]
Got it! So, let me clean up the commits. Do you want me to do something else here?

**Comment by Kampfkarren** [comment]
Don't believe so--just change the PR name and such to specify the new purpose of this PR, being to add tests.

**Comment by Kampfkarren** [comment]
Thanks!

**Pull request merged**